contents/blog/ab-testing-mistakes.md

```diff
@@ -0,0 +1,172 @@
+function showNewChanges(user) {
+  if (posthog.getFeatureFlag('experiment-key') === 'control') {
+    return false;
+  }
+
+  if (user.hasCompletedAction) {
+    return false
+  }
+
+  // other checks
+
+  return true
+}
+```
+
+```js
+// Correct. Will exclude unaffected users
+function showNewChanges(user) {
+
+  if (user.hasCompletedAction) {
+    return false
+  }
+
+  // other checks
+
+  if (posthog.getFeatureFlag('experiment-key') === 'control') {
+    return false;
+  }
+
+  return true
+}
+```
+
+## 2. Only viewing results in aggregate
+
+It's possible that the results of an experiment can show one outcome when analyzed at an aggregated level, but when analyzing the same data in subgroups, it reveals the opposite outcome. To understand this better, let's look at an example:
+
+Suppose you are testing a change to your sign-up and onboarding flow. The change affects both desktop and mobile users. Your experiment results show the following:
+
+| Variant         | Visitors | Conversions | Conversion Rate |
+|-----------------|----------|-------------|-----------------|
+| Control  | 5,000    | 500         | ❌ 10%             |
+| Test     | 5,000    | 1,000       | ✅ 20%             |
+
+At first glance, the test variant seems to be the clear winner. However, breaking down the results into the desktop and mobile subgroups shows:
+
+| Device           | Variant          | Visitors | Conversions | Conversion Rate |
+|-----------|------------------|----------|-------------|-----------------|
+| 💻 Desktop   | Control   | 2,000    | 400         | ✅ 20%             |
+|           | Test      | 2,000    | 100         | ❌ 5%              |
+| 📱 Mobile    | Control   | 3,000    | 100         | ❌ 10%             |
+|           | Test      | 3,000    | 900         | ✅ 30%             |
+
+It's clear now that while the test variant performed better for mobile users, it actually decreased desktop conversions. However, we missed this insight when we combined these metrics! This phenomenon is known as [Simpson's paradox](https://medium.com/homeaway-tech-blog/simpsons-paradox-in-a-b-testing-93af7a2f3307).
+
+Depending on your app and experiment, here's a list of aggregate metrics you want to breakdown:
+* Device type, e.g., desktop or mobile, iOS or Android.
+* Geographic location.
+* Subscription or pricing tier.
+* Acquisition channel, e.g., organic search, paid ads, or referrals, 
+* User role or job function, e.g., manager, executive, or individual contributor.
+* User tenure
+* Business size, e.g., small, medium, or large.
+
+## 3. Conducting an experiment without a predetermined duration
+
+You may even fall victim to the "[peeking problem](https://gopractice.io/data/peeking-problem/)": when you check the intermediate results for statistical significance, make decisions based on them, and end your experiment too early. Without determining how long your experiment should run, you cannot differentiate between intermediate and final results.
```

Comment:
When I first read this, I thought a sentence was missing from the start. It makes sense on second reading, but it's best you tweak the opening a little so readers aren't thrown off their rhythm.